src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -376,11 +383,33 @@ internal bool ShouldRestartWorkerChannel(string runtime, bool isWebHostChannel,
             return string.Equals(_workerRuntime, runtime, StringComparison.InvariantCultureIgnoreCase) && (isWebHostChannel || isJobHostChannel);
         }
 
-        private async Task RestartWorkerChannel(string runtime, string workerId)
+        private async Task RestartWorkerChannel(string runtime)
         {
+            if (_disposing || _disposed)
+            {
+                return;
+            }
             if (_languageWorkerErrors.Count < ErrorEventsThreshold)
             {
-                await InitializeJobhostLanguageWorkerChannelAsync(_languageWorkerErrors.Count);
+                try
+                {
+                    // Issue only one restart at a time.
+                    await _restartWorkerProcessSLock.WaitAsync();
+                    await InitializeJobhostLanguageWorkerChannelAsync(_languageWorkerErrors.Count);
+                }
+                finally
+                {
+                    // Wait before releasing the lock to give time for the process to startup and initialize.
+                    try
+                    {
+                        await Task.Delay(_restartWait, _disposeToken.Token);
+                        _restartWorkerProcessSLock.Release();
+                    }
+                    catch (TaskCanceledException ex)
+                    {
+                        _logger.LogError($"Restart wait task canceled possibly because host instance is being disposed/recycled", ex);
```

Comment:
if this does work...no need to log error...ignoring is OK!

Reply:
Removed log.